### ISSUE TITLE
Incompatibility with Mkdocs Material Blog plugin

### ISSUE THREAD

**Issue by nathancatania** [ultrabug/mkdocs-static-i18n#283]

**Comment by youyoubilly** [comment]
I'm having the same error, please help! Big thanks!

**Comment by ultrabug** [comment]
Yes this is a known incompatibility unfortunately, mkdocs-material broke support when they rewrote the blog plugin and I've been unable to fix that since.

That's on my TODO list but can't share any ETA tbh.

**Comment by niccokunzmann** [comment]
I just added an `if`:

```
                if file.page:
                    file.page._set_canonical_url(mkdocs_config.get('site_url', None))
```

This makes the build succeed but the blog is still empty.

**Comment by SWHL** [comment]
I have met the same problem.

**Comment by kamilkrzyskow** [comment]
In the Gothic Modding Community we finally decided we need the blog plugin to allow for more flexibility when adding non-documentation pages. We kept putting it off, because of the incompatibility here. However, one of the maintainers mentioned that out page is used primarily by the `default` (English) language users, so why don't we just create a non-i18n default language blog hack. I'm still working on the PR for our project, which you can see here:
- https://github.com/Gothic-Modding-Community/gmc/pull/131

Here is the extracted hook from all of my hacky changes:
<details open>
<summary> Hook </summary>

```py
"""MkDocs hook that hides the blog files from the i18n plugin

MIT Licence 2024 Kamil Krzyśków (HRY) for Gothic Modding Community (https://gothic-modding-community.github.io/)
"""

from mkdocs import plugins
from mkdocs.structure.files import Files


BLOG_FILES = []
"""Store the removed blog files here"""


@plugins.event_priority(-95)
def _on_files_disconnect_blog_files(files, config):
    non_blog_files = []

    for file in files:
        # to keep it simple we just define the blog prefix manually here, not reading it from the config
        if file.src_uri.startswith("blog/"):
            BLOG_FILES.append(file)
        else:
            non_blog_files.append(file)

    return Files(non_blog_files)


# Breaking the convention of a maximal -100 to enable this after the i18n plugin
@plugins.event_priority(-105)
def _on_files_connect_blog_files(files, config):
    # Add the blog files back to generate them later on
    for file in BLOG_FILES:
        files.append(file)

    return files


on_files = plugins.CombinedEvent(_on_files_disconnect_blog_files, _on_files_connect_blog_files)
```

</details>


Here is a mini example I tested:
- [i18n-with-default-blog.zip](https://github.com/user-attachments/files/16128107/i18n-with-default-blog.zip)

**Comment by machinaeXlasse** [comment]
For the blog I am fine with the default language too so this is really great.

Thanks so much for sharing!

**Comment by SWHL** [comment]
@kamilkrzyskow I tried your script, and it can solve the part problem.
Without using the hook script, the blog of my site is empty. After using the hook script, the blog can render correctly.

But it displays blogs in all languages simultaneously. Normally, when I choose English, the page only displays the English version of the blog. I choose Chinese, and the page only displays the Chinese version of the blog.

![image](https://github.com/user-attachments/assets/7a2cf8ba-3d67-4a15-8895-56b6d742351e)

Here is a mini example I tested:
[i18n-with-default-blog.zip](https://github.com/user-attachments/files/16249386/i18n-with-default-blog.zip)

**Comment by kamilkrzyskow** [comment]
Hi @SWHL,
you have misunderstood. The purpose of this hook is to only add the option to add default language posts, and they'll will be added for every language.

Currently the i18n plugin doesn't support the blog plugin at all, so it's empty, no posts are shown. 
This hook allows to add files and make them display. 
So do not add multiple languages, only add one language, preferably the default language.

❗ This hook doesn't fix the lack of i18n support for the blog, via this plugin.

**Comment by SWHL** [comment]
@kamilkrzyskow  Thank you for your reply, I did misunderstand.
